python/cudf/cudf/core/column/string.py

```diff
@@ -2086,6 +2086,12 @@ def to_array(self, fillna=None):
 
         return self.to_arrow().to_pandas().values
 
+    def __array__(self, dtype=None):
+        raise NotImplementedError(
+            "__array__ is not supported in CuDF, use .to_array()\
+                 for converting to numpy instead."
```

Comment:
We should craft this error message better to guide users to use cupy if possible and only call `.to_array()` if they explicitly want to copy to host. Can you take a stab at rewording?

Reply:
Okay, What about the cases we have strings?

Reply:
Detailed the error messages, could you also have a look at the string error message if it sounds good?